tests/functions/test_concatenate_columns.py

```diff
@@ -28,7 +28,7 @@ def test_concatenate_columns_null_values(missingdata_df):
         new_column_name="index",
         ignore_empty=True,
     )
-    expected_values = ["1.0-1", "2.0-2", "nan-3"] * 3
+    expected_values = ["1.0-1", "2.0-2", "3"] * 3
     assert expected_values == df["index"].tolist()
 
 
```

Comment:
I also think it might be worth writing a test merging a custom dataframe with a float column (NaN), a datetime column (NaT) and a string column (None/NA?).

And assert the expected output accordingly.

Then, mention this PR or the attached issue in the test docstring as well, please.